PWGHF/Utils/utilsEvSelHf.h

```diff
@@ -297,6 +344,28 @@ struct HfEventSelection : o2::framework::ConfigurableGroup {
           SETBIT(rejectionMask, EventRejection::Occupancy);
         }
       }
+
+      if (useUpcTrigger) {
+        if (!bcs) {
+          throw std::runtime_error("UPC selection requested but no BC table provided!");
+        }
+        SGCutParHolder sgCuts = setSGPreselection();
+        auto bc = collision.template foundBC_as<BCs>();
+        auto bcRange = udhelpers::compatibleBCs(collision, sgCuts.NDtcoll(), *bcs, sgCuts.minNBCs());
+        auto isSGEvent = sgSelector.IsSelected(sgCuts, collision, bcRange, bc);
+        int issgevent = isSGEvent.value;
+        if (issgevent > EventTypeUpc::DoubleGap) {
+          SETBIT(rejectionMask, EventRejection::UpcEventCut);
+        } else {
+          if (issgevent == EventTypeUpc::SingleGapA) {
+            hUPCollisions->Fill(EventTypeUpc::SingleGapA);
+          } else if (issgevent == EventTypeUpc::SingleGapC) {
+            hUPCollisions->Fill(EventTypeUpc::SingleGapC);
+          } else if (issgevent == EventTypeUpc::DoubleGap) {
+            hUPCollisions->Fill(EventTypeUpc::DoubleGap);
+          }
```

Comment:
Why don't you just fill the `issgevent` value?

Reply:
good point